File: app/src/main/java/net/mmho/photomap2/PhotoViewFragment.java
package net.mmho.photomap2;

import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

public class PhotoViewFragment extends Fragment {

    public static final String EXTRA_IMAGE_ID = "image_id";
    private long image_id;
    private LoadableImageView image = null;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setRetainInstance(true);
        Bundle b = getArguments();
        image_id = b.getLong(EXTRA_IMAGE_ID);
    }

    @Override
    public View onCreateView(LayoutInflater inflater,ViewGroup container,Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.photo_view,container,false);
        image = (LoadableImageView)v.findViewById(R.id.photo_view);
        image.startLoading(getActivity().getSupportLoaderManager(),(int)image_id,image_id,null);
        return v;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        BitmapDrawable drawable = (BitmapDrawable) image.getDrawable();
        if(drawable!=null) {
            Bitmap bmp = drawable.getBitmap();
            if (bmp != null) bmp.recycle();
        }
    }
}

File: app/src/main/kotlin/net/mmho/photomap2/PhotoViewFragment.kt
package net.mmho.photomap2

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.photo_view.view.*

class PhotoViewFragment : Fragment() {
    private var imageId: Long = 0
    private var image: LoadableImageView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        retainInstance = true
        imageId = arguments?.getLong(EXTRA_IMAGE_ID) ?: 0
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        val v = inflater.inflate(R.layout.photo_view, container, false)
        image = v.photo
        image?.load(imageId)
        return v
    }

    override fun onDestroyView() {
        super.onDestroyView()
        image?.setImageDrawable(null)
    }

    companion object {
        const val EXTRA_IMAGE_ID = "imageId"
    }
}
